src/galileo/protect.py

```diff
@@ -0,0 +1,106 @@
+from collections.abc import Sequence
+from typing import Optional, Union
+
+from pydantic import UUID4
+
+from galileo.base import BaseClientModel
+from galileo.constants.protect import TIMEOUT_SECS
+from galileo.resources.api.protect import invoke_protect_invoke_post
+from galileo.resources.models.http_validation_error import HTTPValidationError
+from galileo.resources.models.request import Request as APIRequest
+from galileo.resources.models.response import Response as APIResponse
+from galileo.utils.catch_log import DecorateAllMethods
+from galileo_core.schemas.protect.payload import Payload
+from galileo_core.schemas.protect.request import Request
+from galileo_core.schemas.protect.response import Response
+from galileo_core.schemas.protect.ruleset import Ruleset
+
+
+class Protect(BaseClientModel, DecorateAllMethods):
```

Comment:
Why are we wrapping this in a class?

Reply:
Mainly following the pattern we use in the other modules, but the class also lets us pull in the functionality from `BaseClientModel` and `DecorateAllMethods`